python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/dagster_sling_translator.py

```diff
@@ -0,0 +1,111 @@
+import re
+
+from dagster import AssetKey
+from dagster._annotations import public
+
+
+class DagsterSlingTranslator:
+    @public
+    @classmethod
+    def sanitize_stream_name(cls, stream_name: str) -> str:
+        """A function that takes a stream name from a Sling replication config and returns a
+        sanitized name for the stream.
+
+        By default, this removes any non-alphanumeric characters from the stream name and replaces
+        them with underscores, while removing any double quotes.
+
+        Args:
+            stream_name (str): The name of the stream.
+
+        Examples:
+            Using a custom stream name sanitizer:
+
+            .. code-block:: python
+
+                class CustomSlingTranslator(DagsterSlingTranslator):
+                    @classmethod
+                    def sanitize_stream_name(cls, stream_name: str) -> str:
+                        return stream_name.replace(".", "")
+        """
+        return re.sub(r"[^a-zA-Z0-9_.]", "_", stream_name.replace('"', ""))
+
+    @public
+    @classmethod
+    def get_asset_key(cls, stream_name: str, target_prefix: str = "target") -> AssetKey:
```

Comment:
We use `stream_name` and `target_object` interchangebly here in this PR. Can we standardize on one name?

Reply:
Changed these all to accept the stream definition, should be a lot cleaner now.